test/gtest/mhp/halo-3.cpp

```diff
@@ -0,0 +1,65 @@
+#include "xhp-tests.hpp"
+#include <dr/mhp/views/sliding.hpp>
+
+template <typename T> class Halo : public testing::Test {};
+
+TYPED_TEST_SUITE(Halo, AllTypes);
+
+TYPED_TEST(Halo, suite_works_for_3_processes_only) {
+  EXPECT_EQ(dr::mhp::default_comm().size(), 3); // dr-style ignore
+}
+
+TYPED_TEST(Halo, dv_different_halos) {
+  TypeParam dv(6, dr::mhp::distribution().halo(1, 2));
+
+  iota(dv, 1);
+  dv.halo().exchange();
+
+  if (dr::mhp::default_comm().rank() == 0) {
+    EXPECT_EQ(*(dv.begin() + 2), 3); // from node 1
```

Comment:
all these checks use MPI, not halo regions I suppose

in order to see it create a simple test case named `new_halo_values_are_visible_locally_after_exchange`

1. fill vector with one value (e.g. 7)
2. do call the exchange
3. fill vector with different value (e.g. 8)
4. check halo regions as you do it here and expect 7 there (because exchange was not yet here)
5. then call exchange
6. check halo regions as you do it here and expect 8 there (because now halos are distributed to neighbours)

Such a test should fail in step no 4 because when test uses MPI there will be immediately 8 visible.

Now fix the by adding `local()` call on iterator. E.g. `*(dv.begin()+1).local()` and then your test should start work.

In order to have above test case simplest I advise to use halo(1,1)

After creating such a test fix your existing test cases (add `.local()`)